### ISSUE TITLE
redirect_uri invalid

### ISSUE THREAD

**Issue by tmauricio** [dreampowder/strava_flutter#100]
when opening the website with the login of strava the following message appears:

{"message: "Bad Request", "errors": [{"resource": Application", "field": "redirect_uri", "code": "invalid"}]}

The code is:

Future<void> stravaConnect(context) async {
    final String stravaSecret = AppConfig().config.strava["secret"];
    final String stravaClientId = AppConfig().config.strava["clientId"];
    final stravaClient = StravaClient(secret: stravaSecret, clientId: stravaClientId);

    StravaService(stravaClient).testAuthentication(
      [
        AuthenticationScope.profile_read_all,
        AuthenticationScope.read_all,
        AuthenticationScope.activity_read_all
      ],
      "stravaflutter://redirect"
    ).then((value) {
      print("devolucion login strava");
      print(value);
    });
  }

class StravaService {
  final StravaClient stravaClient;
  StravaService(this.stravaClient) {}

  Future<TokenResponse> testAuthentication(
      List<AuthenticationScope> scopes, String redirectUrl) {
    return stravaClient.authentication.authenticate(
        scopes: scopes,
        redirectUrl: redirectUrl,
        forceShowingApproval: false,
        callbackUrlScheme: "stravaflutter",
        preferEphemeral: true
    );
  }

  Future<void> testDeauthorize() {
    return stravaClient.authentication.deAuthorize();
  }

}

**Comment by dreampowder** [comment]
@tmauricio so i guess you fixed this issue on your own side? 😅

**Comment by tmauricio** [comment]
Yeah, I was able to fix it, it was a layer 8 problem, ha! Thanks anyway.

**Comment by dreampowder** [comment]
that's good news, enjoy!
